### ISSUE TITLE
Unary/binary operators and default literal

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#19848]
@akarpov89 commented on [Mon May 29 2017](https://github.com/dotnet/csharplang/issues/632)

**Version Used**: 
Master branch on May 19
**Steps to Reproduce**:

Compile the following code:

```csharp
class C
{
  void M()
  {
    var a = +default;
    var b = -default;
    var c = ~default;
    var d = !default;
  }
}
```
**Expected Behavior**:
Compiler emits errors CS0023: Operator '+' / '-' / '~' / '!' cannot be applied to operand of type 'default'
**Actual Behavior**:
Code compiles without errors. The local variables have the following types and values:

```ml
a: int = 0
b: int = 0
c: int = -1
d: bool = true
```

I understand that this is an implication of unary operator overload resolution using predefined operator implementations. But I see absolutely no benefit in allowing such a weird code.
The [default literal proposal](https://github.com/dotnet/csharplang/blob/master/proposals/target-typed-default.md) mentions couple of usages that are considered illegal (`throw default` and `default == default`). Could you please consider extending this list?

/cc @jcouv 

---

@jcouv commented on [Mon May 29 2017](https://github.com/dotnet/csharplang/issues/632#issuecomment-304694803)

@akarpov89 Thanks for catching this. I'll look at it this week.
I agree this seems fishy.

**Comment by jcouv** [comment]
LDM discussed this today and we will block `default` as operand to unary and binary operators in C# 7.1. The LDM notes will include more details.
FYI @gafter

**Comment by jcouv** [comment]
Small update: actually `==` and `!=` will still be allowed.

**Comment by jcouv** [comment]
Fixed by #19870
Thanks @akarpov89 for raising the issue.
